devimint/src/tests.rs

```diff
@@ -350,6 +350,151 @@ pub async fn latency_tests(dev_fed: DevFed, r#type: LatencyTest) -> Result<()> {
+            std::env::set_var("FM_FEDIMINTD_BASE_EXECUTABLE", oldest_fedimintd);
+            let fedimintd_version = crate::util::FedimintdCmd::version_or_default().await;
+
+            info!(
+                "running first stress test for fedimintd version: {}",
+                fedimintd_version
+            );
+            let mut dev_fed = dev_fed(process_mgr).await?;
+            let client = dev_fed.fed.new_joined_client("test-client").await?;
+            try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
+
+            // add the current fedimintd binary from PATH
+            paths.push("fedimintd".into());
+            for path in paths.iter().skip(1) {
+                dev_fed.fed.restart_all_with_bin(process_mgr, path).await?;
+                try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
+                let fedimintd_version = crate::util::FedimintdCmd::version_or_default().await;
+                info!(
+                    "### fedimintd passed stress test for version {}",
+                    fedimintd_version
+                );
+            }
+            info!("## fedimintd upgraded all binaries successfully")
+        }
+        UpgradeTest::FedimintCli { mut paths } => {
+            let update_fedimint_cli_path = |path: &PathBuf| {
+                std::env::set_var("FM_FEDIMINT_CLI_BASE_EXECUTABLE", path);
+                let fm_mint_client: String = format!(
+                    "{fedimint_cli} --data-dir {datadir}",
+                    fedimint_cli = crate::util::get_fedimint_cli_path().join(" "),
+                    datadir = crate::vars::utf8(&process_mgr.globals.FM_CLIENT_DIR)
+                );
+                std::env::set_var("FM_MINT_CLIENT", fm_mint_client);
+            };
+
+            anyhow::ensure!(!paths.is_empty(), "Must provide at least 1 binary path");
+            let oldest_fedimint_cli = paths.first().expect("at least 1 binary was provided");
+            update_fedimint_cli_path(oldest_fedimint_cli);
+            let fedimint_cli_version = crate::util::FedimintCli::version_or_default().await;
+
+            info!(
+                "running first stress test for fedimint-cli version: {}",
+                fedimint_cli_version
+            );
+            let dev_fed = dev_fed(process_mgr).await?;
+            let client = dev_fed.fed.new_joined_client("test-client").await?;
+            try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
+
+            // add the current fedimint-cli binary from PATH
+            paths.push("fedimint-cli".into());
+            for path in paths.iter().skip(1) {
+                update_fedimint_cli_path(path);
+                let fedimint_cli_version = crate::util::FedimintCli::version_or_default().await;
+                info!("upgraded fedimint-cli to version: {}", fedimint_cli_version);
+                try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
```

Comment:
incorrect: this creates a new client everytime, you don't use use the old use old datadir.

this tests nothing

Reply:
Good catch! Let me know if you see a more elegant solution than my latest approach.